src/main/kotlin/de/cyface/collector/model/metadata/AttachmentMetaData.kt

```diff
@@ -45,10 +45,11 @@ data class AttachmentMetaData(
      * @param body The request body containing the metadata.
      */
     constructor(body: JsonObject) : this(
-        body.getInteger(FormAttributes.LOG_COUNT.value),
-        body.getInteger(FormAttributes.IMAGE_COUNT.value),
-        body.getInteger(FormAttributes.VIDEO_COUNT.value),
-        body.getLong(FormAttributes.FILES_SIZE.value),
+        // The metadata value are strings, so `body.getInteger()` throws a ClassCastException. Thus, we do:
+        body.getString(FormAttributes.LOG_COUNT.value).toInt(),
+        body.getString(FormAttributes.IMAGE_COUNT.value).toInt(),
+        body.getString(FormAttributes.VIDEO_COUNT.value).toInt(),
+        body.getString(FormAttributes.FILES_SIZE.value).toLong(),
```

Comment:
we were wondering in our discussion why I did this before, but with

`body.getInteger(FormAttributes.LOG_COUNT.value)` I get a ClassCastException, as our client stores **all** metadata attributes as strings. I think the reason for this was, that we transmit the metadata in both, the pre-request body and the upload-request header, and in the header only Strings are supported.

So if we would want to change this, we would also have to refactor all sensor clients and break the upload protocol API => Let's not do this ;-)